test/core-flags/cache-flags.test.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+const { run } = require('../utils/test-utils');
+const { resolve } = require('path');
+
+describe('cahche related flags from core', () => {
+    it('should be successful with --cache ', () => {
+        const { stderr, stdout } = run(__dirname, ['--cache']);
+
+        expect(stderr).toBeFalsy();
+        expect(stdout).toContain(`type: 'memory'`);
+    });
+
+    it('should be successful with --no-cache ', () => {
+        const { stderr, stdout } = run(__dirname, ['--no-cache']);
+
+        expect(stderr).toBeFalsy();
+        expect(stdout).toContain('cache: false');
+    });
+
+    it('should set cache.type', () => {
+        const { stderr, stdout } = run(__dirname, ['--cache-type', 'filesystem']);
+
+        expect(stderr).toBeFalsy();
+        expect(stdout).toContain(`type: 'filesystem'`);
+    });
+
+    it('should set cache.cacheDirectory with --cache-cache-directory', () => {
+        const { stderr, stdout } = run(__dirname, ['--cache-type', 'filesystem', '--cache-cache-directory', '/test-cache-path']);
+        const path = resolve('/test-cache-path');
+
+        expect(stderr).toBeFalsy();
+        if (process.platform === 'win32') {
+            // for windows
+            expect(stdout).toContain('test-cache-path');
+        } else {
+            expect(stdout).toContain(`cacheDirectory: '${path}'`);
+        }
+    });
+
+    it('should set cache.cacheLocation with --cache-cache-locations', () => {
+        const { stderr, stdout } = run(__dirname, ['--cache-type', 'filesystem', '--cache-cache-location', '/test-locate-cache']);
+        const path = resolve('/test-locate-cache');
+
+        expect(stderr).toBeFalsy();
+        if (process.platform === 'win32') {
+            // for windows
+            expect(stdout).toContain('test-locate-cache');
+        } else {
+            expect(stdout).toContain(`cacheLocation: '${path}'`);
+        }
+    });
+
+    it('should set cache.hashAlgorithm with --cache-hash-algorithm', () => {
+        const { stderr, stdout } = run(__dirname, ['--cache-type', 'filesystem', '--cache-hash-algorithm', 'sha256']);
+
+        expect(stderr).toBeFalsy();
+        expect(stdout).toContain(`hashAlgorithm: 'sha256'`);
+    });
+
+    it('should set cache.managedPaths with --cache-managed-paths', () => {
+        const { stderr, stdout } = run(__dirname, ['--cache-type', 'memory', '--cache-managed-paths', '/test-manage-path']);
+
+        expect(stderr).toBeFalsy();
+        if (process.platform === 'win32') {
```

Comment:
We should avoid rely on `process.platform` in tests, can we fix it?

Reply:
Yes